src/data/mystery-encounters/encounters/fun-and-games-encounter.ts

```diff
@@ -175,7 +170,9 @@ async function summonPlayerPokemon(scene: BattleScene) {
     const party = scene.getParty();
     const chosenIndex = party.indexOf(playerPokemon);
     if (chosenIndex !== 0) {
-      [party[chosenIndex], party[0]] = [party[chosenIndex], party[chosenIndex]];
+      const leadPokemon = party[0];
+      party[0] = playerPokemon;
+      party[chosenIndex] = leadPokemon;
```

Comment:
NIT:
```suggestion
[party[0], party[chosenIndex]] = [party[chosenIndex], party[0]];
```
This is a shorthand to switch two items in an array.
I actually found that trick in this project

Reply:
It was broken before because of this